packages/shared/src/components/history/ReadingHistoryList.tsx

```diff
@@ -0,0 +1,75 @@
+import React, { ReactElement } from 'react';
+import classNames from 'classnames';
+import {
+  HideReadHistory,
+  ReadHistoryInfiniteData,
+} from '../../hooks/useReadingHistory';
+import classed from '../../lib/classed';
+import {
+  isDateOnlyEqual,
+  getReadHistoryDateFormat,
+} from '../../lib/dateFormat';
+import ReadingHistoryItem from './ReadingHistoryItem';
+
+const DateTitle = classed('h2', 'typo-body text-theme-label-tertiary');
+
+const getDateGroup = (date: Date) => {
+  const label = getReadHistoryDateFormat(date);
+
+  return (
+    <DateTitle key={label} className={classNames('px-6 my-3 first:mt-0')}>
+      {label}
+    </DateTitle>
+  );
+};
+
+export interface ReadHistoryListProps {
+  data: ReadHistoryInfiniteData;
+  onHide: HideReadHistory;
+  infiniteScrollRef: (node?: Element | null) => void;
+}
+
+function ReadHistoryList({
+  data,
+  onHide,
+  infiniteScrollRef,
+}: ReadHistoryListProps): ReactElement {
+  let currentDate: Date;
+
+  const renderList = () =>
```

Comment:
Maybe we should consider memoizing this. This array can get pretty big

Reply:
I agree. This is a must.